ui/src/app/components/apps-panel/expanded-app.tsx

```diff
@@ -160,18 +166,24 @@ const CromwellButtonRow = (props: {
   );
 };
 
-const RStudioButtonRow = (props: {
+interface rstudioProps extends NavigationProps {
   userApp: UserAppEnvironment;
-  workspaceNamespace: string;
-}) => {
-  const { userApp, workspaceNamespace } = props;
-
+  workspace: WorkspaceData;
+}
+const RStudioButtonRow = fp.flow(
+  withNavigation,
+  withCurrentWorkspace()
+)((props: rstudioProps) => {
+  const {
+    userApp,
+    workspace: { namespace, id },
+    navigate,
+  } = props;
```

Comment:
It's a little bit shorter to write it like this

```suggestion
)(({
    userApp,
    workspace: { namespace, id },
    navigate,
  }: rstudioProps) => {
```